torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -529,10 +540,20 @@ class ShapePropagator {
         return;
       }
       case aten::_unwrap_optional: {
+        // if we have None as input, we need to leave the output alone
         auto input_ivalue = toIValue(node->input());
         if (input_ivalue && input_ivalue->isNone()) {
```

Comment:
Nit: Could you use the mustBeNone api here instead ?

Reply:
Are you certain that that works? I'm happy to do it, but I do rely on your judgement here, that it covers all the cases we need. (And the tests might show.)

Reply:
Pretty sure. The only additional case that it is covering is prim::AutoGradZero, in which case we're not gaining anything by continuing with the other two cases right?

Reply:
Yeah it looks like a candidate for `mustBeNone()`

Reply:
Done now with the amended mustBeNone.